.gitlab-ci.yml

```diff
@@ -54,6 +54,13 @@ test-rust-stable:
     - time cargo test --verbose --all --features bit-vec,generic-array,derive
     - sccache -s
 
+test-rust-miri:
+  stage:                           test
+  <<:                              *docker-env
+  script:
+    - time cargo +nightly miri test --features bit-vec,generic-array,arbitrary --release
```

Comment:
I think the `cargo +nightly miri` requires miri component on the image,
I installed it with `rustup +nightly component add miri`

Would it be possible to have it in the image somehow ? @TriplEight

Reply:
You are using `paritytech/ci-linux` CI image, which is main for substrate and polkadot. I know `miri` is a great tool and it would be cool to have it at least substrate CI. However, it's up to devs to decide.
The remaining downside of having `miri` installed in the image is it sometimes [lags behind](https://rust-lang.github.io/rustup-components-history/) the latest nightly. I'm not sure if it's _that_ important, but so far we've never had any components that do not follow the latest image. That being said, this is completely normal for `paritytech/sccache-ci-ubuntu`, `paritytech/contracts-ci-linux` and ``paritytech/ink-ci-linux` images.
I'd be really against having it installed on every push here and would definitely support your idea having it in the image, but let's ask @bkchr if they want `miri` test in substrate/polkadot? As an alternative, I could do a separate CI image for this project.

Reply:
To illustrate that "lag behind" point: one cannot simply `rustup +nightly component add miri` in the pipeline and expect it to be installed every time. The CI image is updated nightly and sometimes that command will fail.
```
# rustup +nightly component add miri
error: component 'miri' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel nightly
Sometimes not all components are available in any given nightly.
error: backtrace:
error:    0: error_chain::backtrace::imp::InternalBacktrace::new
   1: rustup::dist::manifestation::Manifestation::update
   2: rustup::toolchain::DistributableToolchain::add_component
   3: rustup::cli::rustup_mode::main
   4: rustup_init::main
   5: std::rt::lang_start_internal::{{closure}}::{{closure}}
             at /rustc/c7087fe00d2ba919df1d813c040a5d47e43b0fe7/src/libstd/rt.rs:52
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/c7087fe00d2ba919df1d813c040a5d47e43b0fe7/src/libstd/sys_common/backtrace.rs:130
   6: main
   7: __libc_start_main
   8: <unknown>
   ```
   Due to `miri` is not compatible with this nightly. As a result, you will have flaky pipelines.